les/downloader/statesync.go

```diff
@@ -585,13 +584,35 @@ func (s *stateSync) process(req *stateReq) (int, error) {
 // processNodeData tries to inject a trie node data blob delivered from a remote
 // peer into the state trie, returning whether anything useful was written or any
 // error occurred.
-func (s *stateSync) processNodeData(blob []byte) (common.Hash, error) {
-	res := trie.SyncResult{Data: blob}
+//
+// If multiple requests correspond to the same hash, this method will inject the
+// blob as a result for the first one only, leaving the remaining duplicates to
+// be fetched again.
```

Comment:
wait what? That sounds like a very suboptimal decision? So we have the data that _would_ fulfill a request, but we choose to _not_ use it, but rather request it from the network again?

Reply:
Ahah, not sure where I copy from.

But I guess it's fine? Seems it's just dead code(state sync in LES)